ot/lowrank.py

```diff
@@ -0,0 +1,298 @@
+        Linear OT loss with the optimal OT 
+    Q : array-like, shape (n_samples_a, r)
+        First low-rank matrix decomposition of the OT plan 
+    R: array-like, shape (n_samples_b, r)
+        Second low-rank matrix decomposition of the OT plan
+    g : array-like, shape (r, )
+        Weight vector for the low-rank decomposition of the OT plan
+    
+        
+    References
+    ----------
+    .. Scetbon, M., Cuturi, M., & Peyré, G (2021).
+        Low-Rank Sinkhorn Factorization. arXiv preprint arXiv:2103.04737.
+
+    '''
+
+    # POT backend
+    nx = get_backend(X_s, X_t)
+    ns, nt = X_s.shape[0], X_t.shape[0]
+
+    # Initialize weights a, b
+    if a is None:
+        a = unif(ns, type_as=X_s)
+    if b is None:
+        b = unif(nt, type_as=X_t)
+
+    # Compute rank (see Section 3.1, def 1)
+    r = rank
+    if rank == "auto":
+        r = min(ns, nt)
+    
+    if alpha == "auto":
+        alpha = 1e-10
+
+    # Dykstra algorithm won't converge if 1/rank < alpha (alpha is the lower bound for 1/rank)
+    # (see "Section 3.2: The Low-rank OT Problem (LOT)" in the paper)
+    if 1/r < alpha :
+        raise ValueError("alpha ({a}) should be smaller than 1/rank ({r}) for the Dykstra algorithm to converge.".format(a=alpha,r=1/rank))
+
+    # Default value for shape tensor parameter in LazyTensor 
+    if shape_plan == "auto":
+        shape_plan = (ns,nt) 
+
+    # Low rank decomposition of the sqeuclidean cost matrix (A, B)
+    M1, M2 = compute_lr_cost_matrix(X_s, X_t, nx=None)
+
+    # Compute gamma (see "Section 3.4, proposition 4" in the paper)
+    L = nx.sqrt(3*(2/(alpha**4))*((nx.norm(M1)*nx.norm(M2))**2) + (reg + (2/(alpha**3))*(nx.norm(M1)*nx.norm(M2)))**2)
+    gamma = 1/(2*L)
+    
+    # Initialize the low rank matrices Q, R, g 
+    Q, R, g = nx.ones((ns,r)), nx.ones((nt,r)), nx.ones(r) 
+    k = 100 # not specified in paper ?
+
+
+
+    # -------------------------- Low rank algorithm ------------------------------
+    # see "Section 3.3, Algorithm 3 LOT" in the paper
+
+    for ii in range(k): 
+        # Compute the C*R dot matrix using the lr decomposition of C
+        CR_ = nx.dot(M2.T, R)
+        CR = nx.dot(M1, CR_) 
+        
+        # Compute the C.t * Q dot matrix using the lr decomposition of C
+        CQ_ = nx.dot(M1.T, Q)
+        CQ = nx.dot(M2, CQ_)
+        
+        diag_g = nx.diag(1/g)
+        
+        eps1 = nx.exp(-gamma*(nx.dot(CR,diag_g)) - ((gamma*reg)-1)*nx.log(Q))
+        eps2 = nx.exp(-gamma*(nx.dot(CQ,diag_g)) - ((gamma*reg)-1)*nx.log(R))
+        omega = nx.diag(nx.dot(Q.T, CR))
+        eps3 = nx.exp(gamma*omega/(g**2) - (gamma*reg - 1)*nx.log(g))
+
+        Q, R, g = LR_Dysktra(eps1, eps2, eps3, a, b, alpha, stopThr, numItermax, warn, nx)
+
+
+
+    # ----------------- Compute lazy_plan, value and value_linear  ------------------
+    # see "Section 3.2: The Low-rank OT Problem" in the paper
+
+    # Compute lazy plan (using LazyTensor class)
+    plan1 = Q 
```

Comment:
use the already existing funcyion :https://pythonot.github.io/master/gen_modules/ot.utils.html#ot.utils.get_lowrank_lazytensor

that take into account the matrix and the diagonal